README.md

```diff
@@ -12,68 +12,129 @@ SPDX-License-Identifier: MIT
+   - Add another key named `Privacy - Speech Recognition Usage Description` with a string value that explains why your app requires the speech recognition capability. This will be presented to the user when the app first attempts to perform speech recognition.
+
+These entries are mandatory for apps that utilize microphone and speech recognition features. Failing to provide them will result in your app being unable to access these features. 
 
 ## Example
 
-...
+`SpeechTestView` provides a demonstration of the capabilities of the Spezi Speech module.
+It showcases the interaction with the [`SpeechRecognizer`](https://swiftpackageindex.com/stanfordspezi/spezispeech/documentation/spezispeech/speechrecognizer) to provide speech-to-text capabilities and the [`SpeechSynthesizer`](https://swiftpackageindex.com/stanfordspezi/spezispeech/documentation/spezispeech/speechsynthesizer) to generate speech from text.
+
 
 ```swift
-...
+struct SpeechTestView: View {
+   // Get the `SpeechRecognizer` and `SpeechSynthesizer` from the SwiftUI `Environment`.
+   @Environment(SpeechRecognizer.self) private var speechRecognizer
+   @Environment(SpeechSynthesizer.self) private var speechSynthesizer
+   // The transcribed message from the user's voice input.
+   @State private var message = ""
+
+
+   var body: some View {
+      VStack {
+         // Button used to start and stop recording by triggering the `microphoneButtonPressed()` function.
+         Button("Record") {
+            microphoneButtonPressed()
+         }
+            .padding(.bottom)
+
+
+         // Button used to start and stop playback of the transcribed message by triggering the `playbackButtonPressed()` function.
+         Button("Playback") {
+            playbackButtonPressed()
+         }
+      }
+   }
+
+
+   private func microphoneButtonPressed() {
+      if speechRecognizer.isRecording {
+         // If speech is currently recognized, stop the transcribing.
+         speechRecognizer.stop()
+      } else {
+         // If the recognizer is idle, start a new recording.
+         Task {
+            do {
+               // The `speechRecognizer.start()` function returns an `AsyncThrowingStream` that yields the transcribed text.
+               for try await result in speechRecognizer.start() {
+                  // Access the string-based result of the transcribed result.
+                  message = result.bestTranscription.formattedString
+               }
+            }
+         }
+      }
+   }
+    
+   private func playbackButtonPressed() {
+      if speechSynthesizer.isSpeaking {
+         // If speech is currently synthezized, pause the playback.
+         speechSynthesizer.pause()
+      } else {
+         // If synthesizer is idle, start with the text-to-speech functionality.
+         speechSynthesizer.speak(message)
+      }
+   }
+}
 ```
 
-For more information, please refer to the API documentation.
```

Comment:
It would be great if we can include this sentence as noted in the documentation guide (https://swiftpackageindex.com/stanfordspezi/spezi/documentation/spezi/documentation-guide#Example) and point it to the link where the docs will be in the Spezi Package Index?